src/main/java/com/google/devtools/build/lib/starlarkbuildapi/RunEnvironmentInfoApi.java

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.starlarkbuildapi;
+
+import com.google.devtools.build.docgen.annot.DocCategory;
+import com.google.devtools.build.docgen.annot.StarlarkConstructor;
+import com.google.devtools.build.lib.starlarkbuildapi.core.ProviderApi;
+import com.google.devtools.build.lib.starlarkbuildapi.core.StructApi;
+import java.util.List;
+import java.util.Map;
+import net.starlark.java.annot.Param;
+import net.starlark.java.annot.ParamType;
+import net.starlark.java.annot.StarlarkBuiltin;
+import net.starlark.java.annot.StarlarkMethod;
+import net.starlark.java.eval.Dict;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.Sequence;
+
+/**
+ * Provider containing any additional environment variables for use when running executables, either
+ * in test actions or when executed via the run command.
+ **/
+@StarlarkBuiltin(
+    name = "RunEnvironmentInfo",
+    category = DocCategory.PROVIDER,
+    doc = "A provider that can be returned from executable rules to control the environment in"
+        + " which their executable is executed.")
+public interface RunEnvironmentInfoApi extends StructApi {
+
+  @StarlarkMethod(
+      name = "environment",
+      doc = "A dict containing environment variables which should be set when executing the rule"
```

Comment:
Does this doc show up somewhere? It'd be nice if it repeated the key/value data types, or referred the reader to the ctor-doc. Similar for "what happens if the same keys are in inherited_environment"

Reply:
I'm not so sure, but just in case, I copied the docs as I don't know whether it's possible to include a linked cross ref.